botorch/models/approximate_gp.py

```diff
@@ -386,6 +371,72 @@ def __init__(
 
         self.to(train_X)
 
+    def init_inducing_points(
+            self,
+            inputs: Tensor,
+    ) -> Tensor:
+        r"""
+        Reinitialize the inducing point locations in-place with the current kernel
+        applied to `inputs`.
+
+        Args:
+            inputs: input_data (*batch_shape, n, d)
+
+        Returns:
+            (*batch_shape, m, d) selected inducing point locations
+        """
+
+        with torch.no_grad():
+            num_inducing = self.model.variational_strategy.base_variational_strategy.inducing_points.size(-2)
+            inducing_points = _select_inducing_points(inputs, self.model.covar_module, num_inducing,
+                                                      self._input_batch_shape)
+            self.model.variational_strategy.base_variational_strategy.inducing_points.copy_(
+                inducing_points.clone()
+            )
+        return inducing_points
+
+
+def _select_inducing_points(
+        inputs: Tensor,
+        covar_module: Module,
+        num_inducing: int,
+        input_batch_shape: torch.Size,
+) -> Tensor:
+    r"""
+    Utility function that evaluates a kernel at `inputs` and selects inducing point
+    locations based on the pivoted Cholesky heuristic.
+
+    Args:
+        inputs: input data (*batch_shape, n, d)
+        covar_module: GPyTorch Module returning a LazyTensor kernel matrix
+        num_inducing: the max number (m) of inducing points (m <= n)
+        input_batch_shape: non-task-related batch shape
+
+    Returns:
+        (*batch_shape, m, d) inducing point locations
+    """
+
+    train_train_kernel = covar_module(inputs).evaluate_kernel()
+
+    # TODO this assumes X is the same along the batch dimensions
+    if train_train_kernel.ndimension() == 2:
+        pass
+    elif train_train_kernel.ndimension() >= 3:
+        for _ in range(train_train_kernel.ndimension() - 2):
+            train_train_kernel = train_train_kernel[0]
```

Comment:
previously we were essentially assuming the kernel matrix was either 2 or 3 dimensional

Reply:
really we probably ought to be initializing the inducing points separately for each batch element